torch_xla/csrc/tensor.cpp

```diff
@@ -1685,4 +1703,28 @@ Device XLATensor::CommonDeviceForTensors(
   return device;
 }
 
+void XLATensor::CheckRank(const XLATensor& t, xla::int64 expected_rank,
+                          const std::string& tag, const std::string& arg_name,
+                          int arg_number) {
+  xla::int64 actual_rank = t.shape().get().rank();
+  XLA_CHECK_EQ(actual_rank, expected_rank)
+      << "Expected " << expected_rank << "-dimensional tensor, but got "
+      << actual_rank << "-dimensional tensor for "
+      << CheckArgumentString(arg_name, arg_number)
```

Comment:
You can just stream "argument" << number here. Seems a bit light of an API, given the only place we are using it is in a streaming context.

Reply:
Done.